posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -398,21 +403,24 @@ async def insert_into_bigquery_activity(inputs: BigQueryInsertInputs) -> Records
         requires_merge = (
             isinstance(inputs.batch_export_model, BatchExportModel) and inputs.batch_export_model.name == "persons"
         )
+        stage_table_name = f"stage_{inputs.table_id}" if requires_merge else inputs.table_id
 
         with bigquery_client(inputs) as bq_client:
             async with (
                 bq_client.managed_table(
                     inputs.project_id,
                     inputs.dataset_id,
-                    f"{inputs.table_id}",
+                    inputs.table_id,
                     schema,
                     delete=False,
                 ) as bigquery_table,
                 bq_client.managed_table(
                     inputs.project_id,
                     inputs.dataset_id,
-                    f"stage_{inputs.table_id}",
+                    stage_table_name,
                     schema,
+                    create=requires_merge,
+                    delete=requires_merge,
```

Comment:
This saves us from requiring delete permissions as `bigquery_stage_table` will be the same as `bigquery_table` if no merge is required.